font/src/ft/mod.rs

```diff
@@ -213,16 +222,25 @@ impl FreeTypeRasterizer {
         let size = Size::new(size.as_f32_pts() * self.device_pixel_ratio * 96. / 72.);
         self.pixel_size = f64::from(size.as_f32_pts());
 
+        let mut pattern = Pattern::new();
+        pattern.add_family(&desc.name);
+        pattern.add_pixelsize(self.pixel_size);
+
+        let hash = pattern.hash();
+
         match desc.style {
             Style::Description { slant, weight } => {
                 // Match nearest font
-                self.get_matching_face(&desc, slant, weight)
+                pattern.set_weight(weight.into_fontconfig_type());
+                pattern.set_slant(slant.into_fontconfig_type());
             },
             Style::Specific(ref style) => {
                 // If a name was specified, try and load specifically that font.
-                self.get_specific_face(&desc, &style)
+                pattern.add_style(style);
             },
         }
+
+        self.query_font(pattern, hash, desc)
```

Comment:
If I understand this correctly, this is just to deduplicate code, right? Seems like we just get rid of two functions that do not actually need to be separate functions.